libbpf-tools/trace_helpers.c

```diff
@@ -1042,7 +1065,7 @@ bool fentry_exists(const char *name, const char *mod)
 err_out:
 	btf__free(btf);
 	btf__free(base);
-	return id > 0;
+	return id > 0 ? fentry_can_attach(id) : false;
```

Comment:
I think this commit will also close #3781 :)

Reply:
It should help, but I don't think all of the tools listed there use fentry_exists to fall back to kprobes